framework/decode/vulkan_rebind_allocator.cpp

```diff
@@ -710,64 +718,94 @@ VkResult VulkanRebindAllocator::BindImageMemory(VkImage
     if ((image != VK_NULL_HANDLE) && (allocator_image_data != 0) && (allocator_memory_data != 0) &&
         (bind_memory_properties != nullptr))
     {
-        VmaAllocation allocation          = VK_NULL_HANDLE;
-        auto          resource_alloc_info = reinterpret_cast<ResourceAllocInfo*>(allocator_image_data);
-        auto          memory_alloc_info   = reinterpret_cast<MemoryAllocInfo*>(allocator_memory_data);
-
-        VkMemoryRequirements requirements;
-        functions_.get_image_memory_requirements(device_, image, &requirements);
+        auto memory_alloc_info = reinterpret_cast<MemoryAllocInfo*>(allocator_memory_data);
 
-        VmaAllocationCreateInfo create_info;
-        create_info.flags = 0;
-        create_info.usage =
-            GetImageMemoryUsage(resource_alloc_info->usage,
-                                resource_alloc_info->tiling,
-                                device_memory_properties.memoryTypes[memory_alloc_info->original_index].propertyFlags,
-                                requirements);
-        create_info.requiredFlags  = 0;
-        create_info.preferredFlags = 0;
-        create_info.memoryTypeBits = 0;
-        create_info.pool           = VK_NULL_HANDLE;
-        create_info.pUserData      = nullptr;
+        if (memory_alloc_info->ahb)
+        {
+            VkImportAndroidHardwareBufferInfoANDROID importAHBInfo;
+            importAHBInfo.sType  = VK_STRUCTURE_TYPE_IMPORT_ANDROID_HARDWARE_BUFFER_INFO_ANDROID;
+            importAHBInfo.pNext  = nullptr;
+            importAHBInfo.buffer = memory_alloc_info->ahb;
+
+            VkMemoryDedicatedAllocateInfo dedicatedAllocateInfo{};
+            dedicatedAllocateInfo.sType = VK_STRUCTURE_TYPE_MEMORY_DEDICATED_ALLOCATE_INFO;
+            dedicatedAllocateInfo.pNext = &importAHBInfo;
+            dedicatedAllocateInfo.image = image;
+
+            VkMemoryAllocateInfo allocate_info{};
+            allocate_info.sType           = VK_STRUCTURE_TYPE_MEMORY_ALLOCATE_INFO;
+            allocate_info.pNext           = &dedicatedAllocateInfo;
+            allocate_info.allocationSize  = memory_alloc_info->allocation_size;
+            allocate_info.memoryTypeIndex = memory_alloc_info->original_index;
+            result                        = functions_.allocate_memory(device_, &allocate_info, nullptr, &memory);
```

Comment:
Would prefer not to reuse the input value `memory` to hold the new memory handle. Also it seems like we are going to leak this handle and never free it.